tsconfig.json

```diff
@@ -0,0 +1,43 @@
+{
+  "compilerOptions": {
+    "target": "ESNext",
+    "module": "ESNext",
+    "allowJs": true,
+    "checkJs": false,
```

Comment:
This seems like something that would be good to turn on and fix errors for, even now (or in a followup PR), unless I'm misunderstanding?

Reply:
There are too much error in js file, I think it would be better to migrate js to ts and fix them in the same time. It will be very hard to fix all js errors in this or a single PR.

Reply:
Sounds good! Sorry for the churn.